packages/wallet-ui/src/services/useStarkNetSnap.ts

```diff
@@ -644,6 +680,22 @@ export const useStarkNetSnap = () => {
     return txStatus;
   };
 
+  const waitForAccountCreation = async (transactionHash: string, accountAddress: string, chainId: string) => {
+    dispatch(enableLoadingWithMessage('Waiting for transaction to be finalised.'));
+    try {
+      // read transaction to check if the txn is ready
+      await waitForTransaction(transactionHash, chainId);
+      await recoverAccounts(chainId);
```

Comment:
it looks like we dont have retry on recoverAccounts like 'waitForAccountUpdate'? what is the reason for not retry ?

is it once the execution in L2, the account will be recover ?

is there any case, the tx complete, but the contract still not deployed?

and if the tx complete, contract deployed, how to kick start the next workflow "upgrade"